[logs] undefined properties state]

Using

```              
clever-tools  3.4.0
Node.js       v18.15.0
Platform      linux
Release       6.7.9+
Architecture  x64
Linux         Exherbo Linux
Shell         /bin/zsh
Packaged      true
Exec path     /usr/x86_64-pc-linux-gnu/bin/clever
Config file   /home/kannar/.config/clever-cloud/clever-tools.json
Auth source   configuration file
```

I've got a weird error when Ctrl + C the clever logs flow:

```
2024-03-29T12:47:32.202Z: [success] Total time: 239 s (03:59), completed Mar 29, 2024, 12:47:32 PM
2024-03-29T12:47:47.459Z: Creating build cache archive
2024-03-29T12:47:52.384Z: build cache archive successfully created
2024-03-29T12:47:52.384Z: No cron to setup
2024-03-29T12:47:52.405Z: Uploading application build cache archive… file is 2.0G before compression.
2024-03-29T12:47:58.659Z: 2024-03-29T12:47:58.659445Z  INFO multipart_upload_lib::uploader: Uploaded part=1
2024-03-29T12:48:04.533Z: 2024-03-29T12:48:04.533778Z  INFO multipart_upload_lib::uploader: Uploaded part=2
2024-03-29T12:48:10.117Z: 2024-03-29T12:48:10.117483Z  INFO multipart_upload_lib::uploader: Uploaded part=3
2024-03-29T12:48:20.674Z: 2024-03-29T12:48:20.674137Z  INFO multipart_upload_lib::uploader: Uploaded part=4
2024-03-29T12:48:27.159Z: 2024-03-29T12:48:27.159000Z  INFO multipart_upload_lib::uploader: Uploaded part=5
^C[ERROR] Cannot read properties of undefined (reading 'state')
```

--- a/src/models/log-v4.js
+++ b/src/models/log-v4.js
@@ -133,6 +133,11 @@ export async function watchDeploymentAndDisplayLogs(options) {
     logsStream.close(quiet ? 'quiet' : 'follow');
   }
 
+  // deploymentEnded can be undefined if deferred resolved (e.g., stream closed via SIGINT)
+  if (deploymentEnded == null) {
+    return;
+  }
+
   if (deploymentEnded.state === 'OK') {
     Logger.println('');
 

--- a/src/models/log.js
+++ b/src/models/log.js
@@ -133,6 +133,11 @@ export async function watchDeploymentAndDisplayLogs({
     logsStream.close();
   }
 
+  // deploymentEnded can be undefined if deferred resolved (e.g., stream closed via SIGINT)
+  if (deploymentEnded == null) {
+    return;
+  }
+
   if (deploymentEnded.state === 'OK') {
     Logger.println(styleText(['bold', 'green'], 'Deployment successful'));
   } else if (deploymentEnded.state === 'CANCELLED') {

